### DISCUSSION TITLE
Guru Meditation on Sprite Scrolling.

### DISCUSSION THREAD

**Discussion by FerFemNemBem** [Bodmer/TFT_eSPI#1724] [Q&A - General]
Hi,

I have a sprite with a long text inside (171 characters) that I want to scroll. The call of
`scrollinfo.scroll(-1);`
ends in the following "Guru Meditation":
```
Guru Meditation Error: Core  1 panic'ed (LoadProhibited). Exception was unhandled.
Core 1 register dump:
PC      : 0x4000c3df  PS      : 0x00060330  A0      : 0x800d709c  A1      : 0x3ffcd270  
A2      : 0x00000001  A3      : 0x00000001  A4      : 0x000007bf  A5      : 0x00000008
A6      : 0x7ff00000  A7      : 0x7ffe0000  A8      : 0x00000000  A9      : 0x000007bf  
A10     : 0x00000402  A11     : 0x40086db0  A12     : 0x00000012  A13     : 0x3ffef7c3
A14     : 0x00000000  A15     : 0xff000000  SAR     : 0x0000001b  EXCCAUSE: 0x0000001c  
EXCVADDR: 0x000007bf  LBEG    : 0x400014fd  LEND    : 0x4000150d  LCOUNT  : 0xffffffff

ELF file SHA256: 0000000000000000

Backtrace: 0x4000c3df:0x3ffcd270 0x400d7099:0x3ffcd290 0x400d1d27:0x3ffcd2e0 0x400dd164:0x3ffcd330 0x40090992:0x3ffcd350

Rebooting...
ets Jun  8 2016 00:22:57

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0018,len:4
load:0x3fff001c,len:1044
load:0x40078000,len:10124
load:0x40080400,len:5828
entry 0x400806a8
```
If I shorten the String to scroll to e.g. 57 character everything works as expected.
So this looks to me like there is not enough memory available for that big sprite to scroll. 
**But** - if I print out the available memory in both cases right before calling `scrollinfo.scroll(-1);`:
```
SpriteWidth: 1985
HeapSize   : 270400
FreeHeap   : 82444
```
```
SpriteWidth: 680
HeapSize   : 270380
FreeHeap   : 61892
```
than I can see that there is less free heap in case of the smaller string.

So is this really a memory issue? Is there anything I can try to get the scrolling of the long string working?

Best regards,
FFNB

**Comment by Bodmer** [comment]
The free heap is smaller with the smaller sprite idth which is not what would be expected. I suspect the sprite is too bit so space could not be allocated and so scrolling failed. Post an example sketch that demonstrates the problem.

A lower memory consumption approach owuld be to draw one character in a sprite of the screen width, scroll sprite left and then draw the next character etc etc. This would be fast.
